### ISSUE TITLE
ios opencvframework build fail

### ISSUE THREAD

**Issue by tsingjinyun** [opencv/opencv#10363]
```
-- FP16 is not supported by C++ compiler
CMake Error at cmake/OpenCVCompilerOptimizations.cmake:412 (message):
  Required baseline optimization is not supported: NEON
  (CPU_BASELINE_REQUIRE=;NEON)
Call Stack (most recent call first):
  cmake/OpenCVCompilerOptions.cmake:261 (include)
  CMakeLists.txt:519 (include)


-- Performing Test HAVE_CPU_BASELINE_FLAGS
-- Performing Test HAVE_CPU_BASELINE_FLAGS - Failed
CMake Error at cmake/OpenCVCompilerOptimizations.cmake:459 (message):
  Compiler doesn't support baseline optimization flags:
Call Stack (most recent call first):
  cmake/OpenCVCompilerOptions.cmake:264 (ocv_compiler_optimization_options)
  CMakeLists.txt:519 (include)


-- Configuring incomplete, errors occurred!
See also "/Users/Fred/Desktop/dyx_ios/build/build-armv7-iphoneos/CMakeFiles/CMakeOutput.log".
See also "/Users/Fred/Desktop/dyx_ios/build/build-armv7-iphoneos/CMakeFiles/CMakeError.log".
============================================================
ERROR: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DIOS_ARCH=armv7', '-DCMAKE_TOOLCHAIN_FILE=/Users/Fred/Desktop/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DENABLE_NEON=ON', '/Users/Fred/Desktop/opencv', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1
============================================================
Traceback (most recent call last):
  File "platforms/ios/build_framework.py", line 112, in build
    self._build(outdir)
  File "platforms/ios/build_framework.py", line 104, in _build
    self.buildOne(t[0], t[1], mainBD, cmake_flags)
  File "platforms/ios/build_framework.py", line 186, in buildOne
    execute(cmakecmd, cwd = builddir)
  File "platforms/ios/build_framework.py", line 36, in execute
    retcode = check_call(cmd, cwd = cwd)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['cmake', '-GXcode', '-DAPPLE_FRAMEWORK=ON', '-DCMAKE_INSTALL_PREFIX=install', '-DCMAKE_BUILD_TYPE=Release', '-DIOS_ARCH=armv7', '-DCMAKE_TOOLCHAIN_FILE=/Users/Fred/Desktop/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneOS_Xcode.cmake', '-DENABLE_NEON=ON', '/Users/Fred/Desktop/opencv', '-DCMAKE_C_FLAGS=-fembed-bitcode', '-DCMAKE_CXX_FLAGS=-fembed-bitcode']' returned non-zero exit status 1
```

**Comment by tsingjinyun** [comment]
@alalek

**Comment by alalek** [comment]
We don't observe this problem on our build system: http://pullrequest.opencv.org/buildbot/waterfall?tag=ios_pack
You need to investigate reason of this problem yourself (start from comparing build logs and passed parameters).

**Comment by lishishuang1992** [comment]
I also have same this problem, Xcode 9.2    cmake 3.6.3     Mac os 10.12.6  opencv3.2

**Comment by joelteply** [comment]
new to latest command line tools

**Comment by foundry** [comment]
The issue is fixed with  https://github.com/opencv/opencv/pull/10193 (merged after 3.3.1 and before 3.4.0)

**Comment by IvyGongoogle** [comment]
meet the same problem. opencv3.4.5 get works, rather than 3.4.0/3.4.1
